client/landing/subscriptions/components/site-subscription-page/site-subscription-details.tsx

```diff
@@ -51,7 +53,9 @@ const SiteSubscriptionDetails = ( {
 				'You currently have paid subscriptions with this site. Paid subscriptions must be cancelled separately by clicking the Manage Subscriptions button or going to https://wordpress.com/me/purchases.\n\nPress OK to proceed with unsubscribing from the site.\nPress Cancel to go back.'
 			)
 		) {
-			unsubscribe( { blog_id: blogId, url } );
+			const query = new URLSearchParams( location.search );
+			const emailId = query.get( 'email_id' ) || undefined;
+			unsubscribe( { blog_id: blogId, url, emailId } );
```

Comment:
The version with `getQueryArgs()`.
```suggestion
			const emailId = getQueryArgs()?.email_id;
			unsubscribe( { blog_id: blogId, url, emailId } );
```